core/common/src/main/java/alluxio/wire/TieredIdentity.java

```diff
@@ -194,9 +203,37 @@ public boolean equals(Object o) {
       if (!(o instanceof LocalityTier)) {
         return false;
       }
-      LocalityTier that = (LocalityTier) o;
-      return mTierName.equals(that.mTierName)
-          && Objects.equal(mValue, that.mValue);
+      LocalityTier otherTier = (LocalityTier) o;
+      String otherTierName = otherTier.getTierName();
+      if (!mTierName.equals(otherTierName)) {
+        return false;
+      }
+      String otherTierValue = otherTier.getValue();
+      if (mValue == null && otherTierValue == null) {
+        return true;
+      }
+      if (mValue != null && mValue.equals(otherTierValue)) {
+        return true;
+      }
+      // For node tiers, attempt to resolve hostnames to IP addresses, this avoids common
+      // misconfiguration errors where a worker is using one hostname and the client is using
+      // another.
+      if (Constants.LOCALITY_NODE.equals(mTierName)) {
```

Comment:
Could we add a configuration property that allows users to turn this check off? There may be situations where the host resolutions affect performance. Configuration properties are all stored in `PropertyKey.java`

Reply:
Sure, do you have a suggestion for the name of the key?

Reply:
`alluxio.compare.nodenames.by.ip`?

Reply:
```alluxio.compare.nodenames.ip```?